File: LevelSDKExampleApp/app/src/main/java/com/theshopatvsp/levelandroidsdk/ble/model/response/CodePacket.java
package com.theshopatvsp.levelandroidsdk.ble.model.response;

import  com.theshopatvsp.levelandroidsdk.ble.model.response.DataPacket;

/**
 * Created by andrco on 5/27/16.
 */
public class CodePacket extends DataPacket {
    private int code;


    public CodePacket(int code) {
        super();
        this.code = code;
    }

    public CodePacket(byte bytes[]) {
        this.code = bytes[0];
    }

    public byte[] getPacket() {
        byte bytes[] = new byte[1];

        bytes[0] = (byte) code;

        return bytes;
    }

    public int getCode() {
        return code;
    }
}

File: LevelIOSSdkExample/LevelIOSSdkExample/LevelIOSSdk/model/CodePacket.swift
//
// Created by Andrew Cook on 6/24/16.
// Copyright (c) 2016 TheShop. All rights reserved.
//

import Foundation

class CodePacket: DataPacket {
    var code: Int
    
    override init() {
        self.code = 0
        super.init()
    }

    convenience init(code: Int) {
        self.init()
        
        self.code = code
    }

    convenience init(bytes: [UInt8]) {
        self.init(code: Int(bytes[0]))
    }
    
    override func getPacket() -> [UInt8] {
        var bytes: [UInt8] = [UInt8]()
        
        bytes.append(UInt8(code))
        
        return bytes
    }
}
